contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,690 @@
+        require(_owner != address(0), "OTCWrapper: owner address cannot be 0");
+        require(_beneficiary != address(0), "OTCWrapper: beneficiary address cannot be 0");
+        require(_fillDeadline > 0, "OTCWrapper: fill deadline cannot be 0");
+        require(_usdc != address(0), "OTCWrapper: usdc address cannot be 0");
+
+        __Ownable_init();
+        __ReentrancyGuard_init();
+
+        addressbook = AddressBookWrapperInterface(_addressBook);
+        marginRequirements = MarginRequirementsWrapperInterface(addressbook.getMarginRequirements());
+        controller = ControllerWrapperInterface(addressbook.getController());
+        oracle = OracleWrapperInterface(addressbook.getOracle());
+        whitelist = WhitelistWrapperInterface(addressbook.getWhitelist());
+
+        beneficiary = _beneficiary;
+        fillDeadline = _fillDeadline;
+
+        USDC = _usdc;
+    }
+
+    /************************************************
+     *  SETTERS
+     ***********************************************/
+
+    /**
+     * @notice sets the floor and cap of a particular asset notional
+     * @dev can only be called by owner
+     * @param _underlying underlying asset address
+     * @param _min minimum notional value allowed in USD with 6 decimals
+     * @param _max maximum notional value allowed in USD with 6 decimals
+     */
+    function setMinMaxNotional(
+        address _underlying,
+        uint256 _min,
+        uint256 _max
+    ) external onlyOwner {
+        require(_underlying != address(0), "OTCWrapper: asset address cannot be 0");
+        require(_min > 0, "OTCWrapper: minimum notional cannot be 0");
+        require(_max > 0, "OTCWrapper: maximum notional cannot be 0");
+
+        minMaxNotional[_underlying].min = _min;
+        minMaxNotional[_underlying].max = _max;
+    }
+
+    /**
+     * @notice sets the whitelist status of market maker addresses
+     * @dev can only be called by owner
+     * @param _marketMakerAddress address of the market maker
+     * @param _isWhitelisted bool with whitelist status
+     */
+    function setWhitelistMarketMaker(address _marketMakerAddress, bool _isWhitelisted) external onlyOwner {
+        require(_marketMakerAddress != address(0), "OTCWrapper: market maker address cannot be 0");
+        require(
+            isWhitelistedMarketMaker[_marketMakerAddress] != _isWhitelisted,
+            "OTCWrapper: whitelist status already in place"
+        );
+
+        isWhitelistedMarketMaker[_marketMakerAddress] = _isWhitelisted;
+    }
+
+    /**
+     * @notice sets the fee for a given the underlying asset
+     * @dev can only be called by owner
+     * @param _underlying underlying asset address
+     * @param _fee fee amount in bps (100 = 1%)
+     */
+    function setFee(address _underlying, uint256 _fee) external onlyOwner {
+        require(_underlying != address(0), "OTCWrapper: asset address cannot be 0");
+        require(_fee > 0, "OTCWrapper: fee cannot be 0");
```

Comment:
lets not have "fee cannot be 0" - who knows

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/commits/19f4ec38df75ed61239d7b9a96fca4db668b8d0b